### ISSUE TITLE
New-style attrs decorators not recognized as data class decorators

### ISSUE THREAD

**Issue by akirchhoff-modular** [pylint-dev/pylint#9345]
### Bug description

Beginning with attrs 21.1.0, the recommended way to use attrs is through `import attrs` and using `attrs.define`/`attrs.frozen`, not `import attr` and `attr.s` or `attr.attrs`. Pylint does understand `attr.attrs` (#2988), but new-style uses of attrs are not understood to be data class decorators.

```python
"""Example module"""

from attrs import define


@define
class Coordinates:
    """An integer X/Y coordinate on the Cartesian coordinate plane."""

    x: int
    y: int
```

### Command used

```shell
pylint example.py
```

### Pylint output

```python
************* Module example
example.py:7:0: R0903: Too few public methods (0/2) (too-few-public-methods)
```

### Expected behavior

I expect the code to pass pylint without triggering the `too-few-public-methods` check.

### Pylint version

```shell
pylint 3.0.3
astroid 3.0.2
Python 3.9.6 (default, Oct 18 2022, 12:41:40) 
[Clang 14.0.0 (clang-1400.0.29.202)]
```

### Additional dependencies

```python
attrs==23.2.0
```
